velox/vector/ComplexVector.cpp

```diff
@@ -188,8 +183,14 @@ void RowVector::copy(
 
     auto rowSource = source->loadedVector()->as<RowVector>();
     for (auto i = 0; i < childrenSize_; ++i) {
-      children_[i]->copy(
-          rowSource->childAt(i)->loadedVector(), nonNullRows, toSourceRow);
+      if (rowSource->childAt(i)) {
+        BaseVector::ensureWritable(
+            rows, type()->asRow().childAt(i), pool(), children_[i]);
+        children_[i]->copy(
+            rowSource->childAt(i)->loadedVector(), nonNullRows, toSourceRow);
+      } else {
+        children_[i].reset();
+      }
```

Comment:
Wondering in what case would rowSource->childAt(i) be false? And what does it mean if we do children_[i].reset()? Ditto for the two changes below.

Reply:
This is happening because in the expression fuzzer we can have a pre-allocated result vector as nullptr. I am wrapping all preallocated results in a RowVector so that it can re-use the current mechanism of being serialized out to disk on error without changing the format. This would allow older files that directly store a single result vector to still be used correctly by the ExpressionRunner. This means that some children in the RowVector can be null. I reached out to Orri and he mentioned that RowVectors that are referenced might have null children but not those that are copied since they would have been completely loaded, so thats probably why we havent seen this yet in the wild.

children_[i].reset() would just ensure that if the rowSource has a null child then "this" rowVector would also have a null child in that spot.

Reply:
Would it be possible to write a test case around this ?